Add custom suffix for hf mf autopwn to -o argument

zxkmm opened the pull request:
This is because hf mf autopwn is the most used command and this can be used as "taking notes", to prevent causing a not sortable file list after long time use. 

usage:

`hf mf autopwn -o front_door`
`hf mf autopwn -o "front_door"`

iceman1001 commented on CHANGELOG.md:
```diff
@@ -26,6 +26,7 @@ This project uses the changelog in accordance with [keepchangelog](http://keepac
 - Added simulation function to `hf iclass legrec` (@antiklesys)
 - Added keys from Momentum firmware projects. (@onovy)
 - Added Dutch Statistics Agency default key (@eagle00789)
+- Added cuxtom suffix for `hf mf autopwn` command. (@zxkmm)
```
Change to: 

- Changed `hf mf autopwn`  - now allows for custom suffix (@zxkmm)

iceman1001 commented on client/src/cmdhfmf.c:
```diff
@@ -2454,6 +2454,7 @@ static int CmdHF14AMfAutoPWN(const char *Cmd) {
         arg_lit0("a",   NULL,             "Input key A (def)"),
         arg_lit0("b",   NULL,             "Input key B"),
         arg_str0("f", "file",    "<fn>",  "filename of dictionary"),
+        arg_str0("o", "output",  "<fn>",  "filename suffix for dump and key files"),
```
replace    "output"  with  NULL

iceman1001 commented on client/src/cmdhfmf.c:
```diff
@@ -2691,7 +2697,13 @@ static int CmdHF14AMfAutoPWN(const char *Cmd) {
     // read uid to generate a filename for the key file
-    char *fptr = GenerateFilename("hf-mf-", "-key.bin");
+    char suffix[FILE_PATH_SIZE];
+    if (outfilename[0] != '\0') {
```
easier to check  the outfnlen  variable...    `if (outfnlen) `

zxkmm commented:
Thank you for the review!
- fixed changelog
- for the file name buffer i thought check itself is safer but i think check len is safe enough too, so I changed it to yours.
- for the long argument, would you mind tell me why it's better to not define long argument? I already do that as your request but just curious why.

iceman1001 commented on doc/cheatsheet.md:
```diff
@@ -294,6 +294,7 @@ Options:
 -a                             Input key A (def)                                                                
 -b                             Input key B                                                                      
 -f, --file <fn>                filename of dictionary                                                           
+-o, --output <fn>              filename suffix for dump and key files
```
need also to update the text here

iceman1001 commented:
We are trying not to have "long_parameter_names"  since it will fill up out screen space.  Think people using it on phones.   So even if you need a "long" parameter,  try making as short as possible.  In this case  "output"  doesn't add anything.  We could have used  "out" but feels a bit odd.   just making a short -o  is enough since its one of the "reserved" parameter names like -f ..

iceman1001 commented:
Thanks for the fast changes!

zxkmm commented:
> We are trying not to have "long_parameter_names" since it will fill up out screen space. Think people using it on phones. So even if you need a "long" parameter, try making as short as possible. In this case "output" doesn't add anything. We could have used "out" but feels a bit odd. just making a short -o is enough since its one of the "reserved" parameter names like -f ..

Thanks for the explanation!
